src/dawn/engine_dawn.cc

```diff
@@ -417,64 +437,52 @@ Result EngineDawn::DoBuffer(const BufferCommand*) {
   return Result("Dawn:DoBuffer not implemented");
 }
 
-Result EngineDawn::CreateCommandBufferBuilderIfNeeded() {
-  if (command_buffer_builder_)
-    return {};
-
-  // Create the command buffer builder because it doesn't exist yet.
-  if (!device_) {
-    return Result(
-        "EngineDawn: Can't create command buffer builder: device is not "
-        "initialized");
-  }
-  command_buffer_builder_ = device_->CreateCommandEncoder();
-  if (command_buffer_builder_)
-    return {};
-  return Result("EngineDawn: Can't create command buffer builder");
+Result EngineDawn::CreateRenderObjectsIfNeeded(
+    RenderPipelineInfo* render_pipeline) {
+  return CreateFramebufferIfNeeded(render_pipeline);
 }
 
-void EngineDawn::DestroyCommandBufferBuilder() {
-  command_buffer_builder_ = ::dawn::CommandEncoder();
-}
+Result EngineDawn::CreateFramebufferIfNeeded(
+    RenderPipelineInfo* render_pipeline) {
+  Result result;
 
-Result EngineDawn::CreateRenderObjectsIfNeeded() {
-  Result result = CreateCommandBufferBuilderIfNeeded();
-  if (!result.IsSuccess())
-    return result;
-  return CreateFramebufferIfNeeded();
-}
+  const uint32_t width = render_pipeline->pipeline->GetFramebufferWidth();
+  const uint32_t height = render_pipeline->pipeline->GetFramebufferHeight();
 
-Result EngineDawn::CreateFramebufferIfNeeded() {
-  Result result;
+  // First make the Dawn color attachment textures that the render pipeline will
+  // write into.
   {
     ::dawn::Texture fb_texture;
-    result = MakeFramebufferTexture(*device_, kFramebufferFormat, &fb_texture);
+
+    result = MakeFramebufferTexture(*device_, kFramebufferFormat, width, height,
```

Comment:
The framebuffer format should come from the pipeline, the format is specified in the script.

Reply:
done in most recent commit.  This is the biggest change to what you've already seen.